pkg/cookies/cookies.go

```diff
@@ -81,3 +68,19 @@ func ParseSameSite(v string) http.SameSite {
 		panic(fmt.Sprintf("Invalid value for SameSite: %s", v))
 	}
 }
+
+// WarnInvalidDomain logs a warning if the request host and cookie domain are
+// mismatched.
+func WarnInvalidDomain(c *http.Cookie, req *http.Request) {
```

Comment:
Does this need to be exported?

Reply:
Good catch, doesn't look like it anymore. I think there was some duplicated CSRF cookie code directly in oauthproxy.go in earlier commits of this PR that I refactored and consolidated.

I'll make this private now.